### ISSUE TITLE
Test failure in TestAtomicCoverpkgAll

### ISSUE THREAD

**Issue by johanbrandhorst** [neelance/go#6]
After the fix for #5, I'm instead seeing the following error now:

```
--- FAIL: TestAtomicCoverpkgAll (1.51s)
        go_test.go:5737: running testgo [test -coverpkg=all -covermode=atomic x]
        go_test.go:5737: standard output:
        go_test.go:5737: ok     x       0.007s  coverage: 0.0% of statements in all

        go_test.go:5739: running testgo [test -coverpkg=all -race x]
        go_test.go:5739: standard output:
        go_test.go:5739: signal: segmentation fault (core dumped)
                FAIL    x       0.103s

        go_test.go:5739: go [test -coverpkg=all -race x] failed unexpectedly: exit status 1
FAIL
FAIL    cmd/go  54.786s
```
Again, confirmed with a rerun.

**Comment by neelance** [comment]
What command are you running?

**Comment by johanbrandhorst** [comment]
`./all.bash`

**Comment by neelance** [comment]
Without any `GOARCH` set, right? Which OS do you have?

**Comment by johanbrandhorst** [comment]
Yes, just `./all.bash`. I'm running on Linux amd64. I'll get you a full log, just rerunning for the 3rd time as I lost the console log.

**Comment by neelance** [comment]
Could you please try if you see the same with `b77aad089176ecab971d3a72f0357ed0abc81f4a`? That's the commit it is rebased on.

**Comment by neelance** [comment]
Oh wait, I can reproduce this time!

**Comment by johanbrandhorst** [comment]
Oh, sweet! Here's the console log anyway:

```
$ ./all.bash
Building Go cmd/dist using /usr/local/go/.
Building Go toolchain1 using /usr/local/go/.
Building Go bootstrap cmd/go (go_bootstrap) using Go toolchain1.
Building Go toolchain2 using go_bootstrap and Go toolchain1.
Building Go toolchain3 using go_bootstrap and Go toolchain2.
Building packages and commands for linux/amd64.

##### Testing packages.
ok      archive/tar     0.078s
ok      archive/zip     1.510s
ok      bufio   0.117s
ok      bytes   1.359s
ok      compress/bzip2  0.092s
ok      compress/flate  1.301s
ok      compress/gzip   0.065s
ok      compress/lzw    0.008s
ok      compress/zlib   0.057s
ok      container/heap  0.015s
ok      container/list  0.004s
ok      container/ring  0.013s
ok      context 0.990s
ok      crypto  0.012s
ok      crypto/aes      0.053s
ok      crypto/cipher   0.008s
ok      crypto/des      0.028s
ok      crypto/dsa      0.012s
ok      crypto/ecdsa    0.179s
ok      crypto/elliptic 0.037s
ok      crypto/hmac     0.018s
ok      crypto/md5      0.004s
ok      crypto/rand     0.060s
ok      crypto/rc4      0.083s
ok      crypto/rsa      0.073s
ok      crypto/sha1     0.040s
ok      crypto/sha256   0.022s
ok      crypto/sha512   0.013s
ok      crypto/subtle   0.030s
ok      crypto/tls      0.891s
ok      crypto/x509     2.105s
ok      database/sql    0.600s
ok      database/sql/driver     0.017s
ok      debug/dwarf     0.029s
ok      debug/elf       0.016s
ok      debug/gosym     0.223s
ok      debug/macho     0.008s
ok      debug/pe        0.011s
ok      debug/plan9obj  0.004s
ok      encoding/ascii85        0.031s
ok      encoding/asn1   0.012s
ok      encoding/base32 0.030s
ok      encoding/base64 0.021s
ok      encoding/binary 0.019s
ok      encoding/csv    0.022s
ok      encoding/gob    0.040s
ok      encoding/hex    0.023s
ok      encoding/json   0.606s
ok      encoding/pem    0.015s
ok      encoding/xml    0.022s
ok      errors  0.010s
ok      expvar  0.020s
ok      flag    0.003s
ok      fmt     0.103s
ok      go/ast  0.004s
ok      go/build        0.166s
ok      go/constant     0.024s
ok      go/doc  0.038s
ok      go/format       0.008s
ok      go/importer     0.110s
ok      go/internal/gccgoimporter       0.007s
ok      go/internal/gcimporter  0.429s
ok      go/internal/srcimporter 0.737s
ok      go/parser       0.111s
ok      go/printer      0.435s
ok      go/scanner      0.003s
ok      go/token        0.036s
ok      go/types        0.982s
ok      hash    0.008s
ok      hash/adler32    0.038s
ok      hash/crc32      0.023s
ok      hash/crc64      0.007s
ok      hash/fnv        0.003s
ok      html    0.027s
ok      html/template   0.055s
ok      image   0.135s
ok      image/color     0.024s
ok      image/draw      0.123s
ok      image/gif       0.454s
ok      image/jpeg      0.287s
ok      image/png       0.030s
ok      index/suffixarray       0.041s
ok      internal/cpu    0.005s
ok      internal/poll   0.021s
ok      internal/singleflight   0.037s
ok      internal/trace  1.548s
ok      io      0.039s
ok      io/ioutil       0.011s
ok      log     0.044s
ok      log/syslog      1.259s
ok      math    0.003s
ok      math/big        2.493s
ok      math/bits       0.004s
ok      math/cmplx      0.009s
ok      math/rand       0.371s
ok      mime    0.012s
ok      mime/multipart  0.501s
ok      mime/quotedprintable    0.188s
ok      net     2.079s
ok      net/http        3.988s
ok      net/http/cgi    0.906s
ok      net/http/cookiejar      0.034s
ok      net/http/fcgi   0.061s
ok      net/http/httptest       0.048s
ok      net/http/httptrace      0.012s
ok      net/http/httputil       0.065s
ok      net/http/internal       0.023s
ok      net/internal/socktest   0.002s
ok      net/mail        0.006s
ok      net/rpc 0.027s
ok      net/rpc/jsonrpc 0.012s
ok      net/smtp        0.014s
ok      net/textproto   0.005s
ok      net/url 0.046s
ok      os      0.566s
ok      os/exec 0.698s
ok      os/signal       4.783s
ok      os/user 0.003s
ok      path    0.002s
ok      path/filepath   0.027s
ok      reflect 0.201s
ok      regexp  0.161s
ok      regexp/syntax   0.570s
ok      runtime 46.685s
ok      runtime/debug   0.074s
ok      runtime/internal/atomic 0.123s
ok      runtime/internal/sys    0.028s
ok      runtime/pprof   3.874s
ok      runtime/pprof/internal/profile  0.029s
ok      runtime/trace   2.093s
ok      sort    0.072s
ok      strconv 0.607s
ok      strings 0.265s
ok      sync    0.402s
ok      sync/atomic     0.048s
ok      syscall 0.043s
ok      testing 1.230s
ok      testing/quick   0.209s
ok      text/scanner    0.006s
ok      text/tabwriter  0.004s
ok      text/template   0.472s
ok      text/template/parse     0.010s
ok      time    2.815s
ok      unicode 0.052s
ok      unicode/utf16   0.002s
ok      unicode/utf8    0.018s
ok      vendor/golang_org/x/crypto/chacha20poly1305     0.069s
ok      vendor/golang_org/x/crypto/chacha20poly1305/internal/chacha20   0.006s
ok      vendor/golang_org/x/crypto/cryptobyte   0.011s
ok      vendor/golang_org/x/crypto/curve25519   0.017s
ok      vendor/golang_org/x/crypto/poly1305     0.002s
ok      vendor/golang_org/x/net/http2/hpack     0.008s
ok      vendor/golang_org/x/net/idna    0.026s
ok      vendor/golang_org/x/net/lex/httplex     0.007s
ok      vendor/golang_org/x/net/nettest 1.110s
ok      vendor/golang_org/x/net/proxy   0.004s
ok      vendor/golang_org/x/text/transform      0.002s
ok      vendor/golang_org/x/text/unicode/norm   0.036s
ok      cmd/addr2line   2.170s
ok      cmd/api 0.079s
ok      cmd/asm/internal/asm    0.387s
ok      cmd/asm/internal/lex    0.022s
ok      cmd/compile     16.429s
ok      cmd/compile/internal/gc 42.075s
ok      cmd/compile/internal/ssa        0.163s
ok      cmd/compile/internal/syntax     0.037s
ok      cmd/compile/internal/test       0.041s [no tests to run]
ok      cmd/compile/internal/types      0.029s
ok      cmd/cover       1.989s
ok      cmd/doc 0.207s
ok      cmd/fix 6.402s
--- FAIL: TestAtomicCoverpkgAll (3.15s)
        go_test.go:5737: running testgo [test -coverpkg=all -covermode=atomic x]
        go_test.go:5737: standard output:
        go_test.go:5737: ok     x       0.029s  coverage: 0.0% of statements in all

        go_test.go:5739: running testgo [test -coverpkg=all -race x]
        go_test.go:5739: standard output:
        go_test.go:5739: signal: segmentation fault (core dumped)
                FAIL    x       0.099s

        go_test.go:5739: go [test -coverpkg=all -race x] failed unexpectedly: exit status 1
FAIL
FAIL    cmd/go  59.917s
ok      cmd/go/internal/cache   0.158s
ok      cmd/go/internal/generate        0.008s
ok      cmd/go/internal/get     0.007s
ok      cmd/go/internal/load    0.005s
ok      cmd/go/internal/work    0.011s
ok      cmd/gofmt       0.135s
ok      cmd/internal/buildid    0.275s
ok      cmd/internal/dwarf      0.020s
ok      cmd/internal/edit       0.023s
ok      cmd/internal/goobj      1.502s
ok      cmd/internal/obj        0.008s
ok      cmd/internal/obj/arm64  0.006s
ok      cmd/internal/obj/x86    0.644s
ok      cmd/internal/objabi     0.026s
ok      cmd/internal/src        0.043s
ok      cmd/internal/test2json  0.231s
ok      cmd/link        1.623s
ok      cmd/link/internal/ld    43.418s
ok      cmd/nm  4.058s
ok      cmd/objdump     2.976s
ok      cmd/pack        2.789s
ok      cmd/trace       0.060s
ok      cmd/vendor/github.com/google/pprof/internal/binutils    0.753s
ok      cmd/vendor/github.com/google/pprof/internal/driver      10.451s
ok      cmd/vendor/github.com/google/pprof/internal/elfexec     0.003s
ok      cmd/vendor/github.com/google/pprof/internal/graph       0.003s
ok      cmd/vendor/github.com/google/pprof/internal/measurement 0.036s
ok      cmd/vendor/github.com/google/pprof/internal/report      0.245s
ok      cmd/vendor/github.com/google/pprof/internal/symbolizer  0.015s
ok      cmd/vendor/github.com/google/pprof/internal/symbolz     0.025s
ok      cmd/vendor/github.com/google/pprof/profile      0.059s
ok      cmd/vendor/github.com/ianlancetaylor/demangle   0.027s
ok      cmd/vendor/golang.org/x/arch/arm/armasm 0.026s
ok      cmd/vendor/golang.org/x/arch/arm64/arm64asm     0.102s
ok      cmd/vendor/golang.org/x/arch/ppc64/ppc64asm     0.002s
ok      cmd/vendor/golang.org/x/arch/x86/x86asm 0.168s
ok      cmd/vet 3.711s
ok      cmd/vet/internal/cfg    0.028s
2018/03/02 22:04:27 Failed: exit status 1
```

**Comment by neelance** [comment]
Found it. Merge error.
